### PR TITLE
refactor: make error messages consistent when design name is incorrect

### PR DESCRIPTION
**Pull request by Double-Twelve**
### The middle of the top bar can directly modify the name of the design
image

### The design name can be changed e by clicking "Save As"
image

**Before updated:**
These two input boxes have methods for validating user input respectively, and the error messages that are prompted are also different.

**After updated:**
Use the same validation method.

### CODE REVIEW

**Review by wwwseven — APPROVED** [review]
LGTM

File: src/utils/checkDesignName.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { DESIGN_NAME_REGULAR_EXPRESSION } from "@/constants/courtData";
+
+const checkDesignName = (name: string, nameList: string[]) => {
+  // when canBeBlank is false, check if name is blank
+  if (name.length === 0) {
```
Comment (tatoMa) [inline src/utils/checkDesignName.ts:5]: Might need the trim here. I see you have trim when you call checkDesignName else where every time.

File: src/components/TopBar/SaveBoard.tsx [hunk 2]
```diff
@@ -43,12 +40,10 @@ const SaveBoard: React.FC = () => {
   const designsData = useStoreSelector((state) => state.courtSpecData.designsData);
   const designNames = useStoreSelector((state) => state.designName.nameList);
   const cancelRef = useRef(null);
-  const [nameCheck, setNameCheck] = useState<string>("exsited");
+  const [nameCheck, setNameCheck] = useState<string>("");
```
Comment (tatoMa) [inline src/components/TopBar/SaveBoard.tsx:43]: If it has an initial value as a string, you don't need to define the type of string anymore, unless you are unsure(maybe need another type other than string later)

File: src/components/TopBar/SaveBoard.tsx [hunk 3]
```diff
@@ -201,6 +174,7 @@ const SaveBoard: React.FC = () => {
           h="44px"
           justifyContent="left"
           onClick={open}
+          data-testid="sava-as"
         >
           <AlertDialog
```
Comment (tatoMa) [inline src/components/TopBar/SaveBoard.tsx:179]: Please, consider extract the whole alert window to a new file.

File: src/components/NavBar/EditorDesignName.tsx [hunk 4]
```diff
@@ -36,6 +40,21 @@ const DesignName = () => {
...
+      const errorMessage = checkDesignName(newDesignName.trim(), nameList);
+      if (errorMessage) {
+        setFeedbackModalOpen(true);
+        setFeedback(errorMessage);
+      } else {
+        dispatch(changeDesignName(newDesignName.trim()));
+      }
+    }
+  };
+
```
Comment (ZhihanChenZack) [inline src/components/NavBar/EditorDesignName.tsx:57]: The name length check seems like part of the check design name function. It seems like the length check can be moved into the checkDesignName().

**Review by ZhihanChenZack — COMMENTED** [review]
Well done!

File: src/components/NavBar/EditorDesignName.tsx [hunk 5]
```diff
@@ -36,6 +40,21 @@ const DesignName = () => {
     );
   };
 
+  const saveNameChange = () => {
+    if (newDesignName.trim().length > DESIGN_NAME_MAX_CHAR_LENGTH) {
```
Comment (ZhihanChenZack) [inline src/components/NavBar/EditorDesignName.tsx:44]: The name length check seems like part of the check design name function. It seems like the length check can be moved into the checkDesignName().

**Review by GLC-coder — APPROVED** [review]
Well done

**Pull request merged**